internal/services/web/app_service_slot_custom_hostname_binding_resource.go

```diff
@@ -0,0 +1,193 @@
+package web
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/web/mgmt/2021-02-01/web"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/locks"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/web/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/web/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+var appServiceSlotCustomHostnameBindingResourceName = "azurerm_app_service_slot_custom_hostname_binding"
+
+func resourceAppServiceSlotCustomHostnameBinding() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceAppServiceSlotCustomHostnameBindingCreate,
+		Read:   resourceAppServiceSlotCustomHostnameBindingRead,
+		Delete: resourceAppServiceSlotCustomHostnameBindingDelete,
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.AppServiceSlotCustomHostnameBindingID(id)
+			return err
+		}),
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"app_service_slot_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.AppServiceSlotID,
+			},
+
+			"hostname": {
+				Type:     pluginsdk.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"ssl_state": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(web.SslStateIPBasedEnabled),
+					string(web.SslStateSniEnabled),
+				}, false),
+				RequiredWith: []string{"thumbprint"},
+			},
+
+			"thumbprint": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+				RequiredWith: []string{"ssl_state"},
+			},
```

Comment:
@jackofallops you've got the most context on the thumbprint changes of late, does this need changing in any way?

Reply:
I could change this like it is done inside `app_service_custom_hostname_binding_resource.go`. What do you think?
https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/web/app_service_custom_hostname_binding_resource.go#L54